localtime/data_source_timestamp.go

```diff
@@ -0,0 +1,76 @@
+package hashicups
+
+import (
+	"time"
+
+	"github.com/araddon/dateparse"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTimestamp() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceTimestampRead,
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"layout": &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"layout_timezone"},
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					t := time.Now()
+
+					if v == t.Format(v) {
```

Comment:
Interesting condition, can definitely see how this would catch _some_ invalid layouts, but I wonder if it correctly handles a layout with some valid tokens and some unparse-able ones. In any case, the point of validation is to catch some static errors not all, and I can't come up with any better error detection here.

Reply:
Yeah this was the best I can come up with. Thought `t.Format(...)` would return another value for error, like most other functions.